java-squid/src/main/java/org/sonar/java/se/ProgramState.java

```diff
@@ -31,12 +31,26 @@
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.Deque;
+import java.util.HashSet;
 import java.util.LinkedList;
 import java.util.List;
 import java.util.Objects;
+import java.util.Set;
 
 public class ProgramState {
 
+  public static class Pop {
```

Comment:
This inner class is a great improvement. It would have been really great if it have benn part of this pull request  in its own commit but ok.